terraform/common/variables.tf

```
@@ -8,4 +8,12 @@ locals {
   primary_zone_name                     = "teaching-vacancies.service.gov.uk"
   secondary_zone_name                   = "teaching-jobs.service.gov.uk"
   service_name                          = "teaching-vacancies"
+  secrets                               = yamldecode(data.aws_ssm_parameter.common_secrets.value)
+  groups = toset([
+    "${local.service_name}-admin",
+    "${local.service_name}-billing",
+    "${local.service_name}-dev",
+    "${local.service_name}-prod",
+    "${local.service_name}-readonly"
+  ])
```

```suggestion
  groups = toset([for k,v in local.secrets : k])
```
You could generate the keys like this(👆🏼) and don't have to edit terraform code if the yaml keys change.

---

Thanks. That's a nice improvement. I've separated out group membership from group policy attachment (as I might want to define a group, even if there are no members). 
I'm now setting the policies for the groups in the locals too

---

Assigning users to groups will become the responsibility of #digital-tools-support